### ISSUE TITLE
Feature Request: Color Settings

### ISSUE THREAD

**Issue by freiit** [Zodiase/meteor-mdl#7]
Like what you did here! :+1: 

I don't know enough internals about mdl, but it should be simple to choose the different colors? Can you add a settings option? http://www.getmdl.io/customize/index.html

**Comment by niranjans** [comment]
+1

**Comment by Zodiase** [comment]
I'd love to do this if it was that trivial. (sign)

For now the best workaround I can think of is adding the stylesheet of your choice manually in the <head> section of your template, which should be appended after all the package content and therefore overwriting the current stylesheet in this package.
This, however, as you can imagine, could double the size of the stylesheets sent to the client, which is not a good thing performance wise.
Or, you could go to the folder of this package and replace the `material.css` with your customized version and it should be good to go.
(Or, the ideal way if you don't mind getting a little headache figuring it out, find out how to overwrite the coloring only and add those code to your project.)

If you ever find better ways of doing this, let me know or create a pull request for your own implementation. Currently the main focus is still making sure everything works.

**Comment by freiit** [comment]
I diff-ed the files. The only difference is really the color code and I found only a few colors that change:
In the original you find
Pink: rgb(255,64,129)
Blue: rgb(63,81,181)
"Black": rgb(66,66,66)
By replacing these codes with a new color you almost get the same css as you would get when you download directly, but unfortunately there are a few places where the "black" is not replaced and there is one strange color, that changes, too:
.mdl-color--primary-dark{background-color:rgb(48,63,159)!important}

If someone wants to play around with this:
sed 's/255,64,129/255,82,82/g;s/63,81,181/3,169,244/g;s/255,255,255/66,66,66/g'

But imho the real solution would be to to understand the build process of the .css out of the .scss-files...

**Comment by Zodiase** [comment]
For anyone who's still looking into this: 
All the available color presets are defined here: [palette.scss](https://github.com/google/material-design-lite/blob/master/src/palette/_palette.scss).
The default build still contains these so which ever color you want, just add the class name and you are good to go.

You could also change the default color variable here: [variables.scss#L85](https://github.com/google/material-design-lite/blob/master/src/_variables.scss#L85) and build your own `material.css` and replace the one in this package (although you would have to do this every time the package is updated...)

Or, would you like individual theme packages?

**Comment by KristerV** [comment]
uh, not very clear.

I want to change the color theme entirely. What's the easiest way to do this?

edit: just downloading the file and saving it works: http://www.getmdl.io/customize/index.html

**Comment by Zodiase** [comment]
@KristerV Yes overriding the colors with a new theme css file sure works. However the officially provided themes are limited and at some point people would want more freedom in terms of customizing color (though I think that color wheel can provide enough variations).
Also doing this would cause about 124KB more css code to be sent to the client and would render the about 302KB css code from this package useless.

**Comment by Zodiase** [comment]
I couldn't find a way to allow users to have custom settings for packages. Does anyone know how to do that?

One approach I thought about was to include all files from [mdl's release branch](https://github.com/google/material-design-lite/tree/release) so you can go into the [`src` folder](https://github.com/google/material-design-lite/tree/release/src) and modify the [`variables`](https://github.com/google/material-design-lite/blob/release/src/_variables.scss) and rebuild. However I think doing that would force you to repeat this process every time this package gets updated since the changes would be overwritten.

I'm also thinking maybe I could separate the css code and js code into two packages, `zodiase:mdl` and `zodiase:mdl-theme-default` for example. Then you can only use the package containing only js code and use your own theme file, or maybe I could create a theme package for all available themes so you don't have to maintain the theme files.

What do you think?

**Comment by Zodiase** [comment]
Hey guys I came up with a solution that somewhat works.

If you want to try it out, clone [the `support_settings` branch](https://github.com/Zodiase/meteor-mdl/tree/support_settings) and add it as a local package to your app.

**To use the new feature, create a `zodiase-mdl.json` file under the root of your app. This file is going to store all settings for this package in the future. For now, let's add some theme settings first.**

The json file should look like this:

``` JavaScript
{
  "theme": {
    "primary": "deep_orange",
    "accent": "green"
  }
}
```

If the above code isn't self-explanatory to you, basically we're specifying the primary and accent color for the theme, where the color names are identical from [MDL's color wheel](http://www.getmdl.io/customize/index.html). So look up the color names and fill in there. And this package will read this json file and use the theme file you specified.

Note that the default values of the primary color and accent color are "indigo" and "pink", respectively. So if you omit the accent color, for example, the theme file will be `"material.deep_orange-pink.min.css"`.

Unfortunately, currently there is a drawback of this implementation. Every time you change this json file, you have to somehow ask Meteor to rebuild this package. One way of doing this is `meteor reset` but be aware that this also resets your MongoDB and perhaps other stuff so be careful. This drawback is because the theme file is added during the build process of the package, and Meteor believes once a package is built, there is no reason to rebuild it. So... Yeah.

Let me know if you have suggestions and comments. If it looks good, also let me know so I can push this change to the next release.

Oh I forgot to mention there is a demo project also [in this branch](https://github.com/Zodiase/meteor-mdl/tree/support_settings/integration-tests/settings) prepared for testing theme settings. Check it out.

**Comment by Zodiase** [comment]
Since another feature is going to depend on this, let's settle with the current design and implementation. This feature will be available in version `1.0.6_3`, which I'll try to release ASAP.
